src/common/getters.ts

```diff
@@ -0,0 +1,251 @@
+//
+// Copyright Inrupt Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+import type { BlankNode, DatasetCore, Literal, NamedNode } from "@rdfjs/types";
+import { DataFactory } from "n3";
+import { isUrl } from "./common";
+import { cred, dc, rdf, sec, xsd } from "./constants";
+import { getSingleObject, lenientSingle } from "./rdfjs";
+
+const { namedNode, defaultGraph, quad } = DataFactory;
+
+export type DatasetWithId = DatasetCore & { id: string };
```

Comment:
The pattern we've used in `solid-client` is to define a `WithId` type, and not necessarily have a `DatasetWithId` but rather a `Datasetcore & WithId`. Arguably, that could be aliased to keep `DatasetWithId`.

This allows to be a bit more fine-grained: for instance, `getId` only needs a `withId` typed argument.